tests/models/soil/test_soil_model.py

```diff
@@ -368,25 +363,80 @@ def test_replace_soil_pools(dummy_carbon_data, soil_model_fixture):
     ],
 )
 def test_integrate_soil_model(
-    mocker, caplog, soil_model_fixture, output, raises, expected_log
+    mocker, caplog, soil_model_fixture, mock_output, raises, final_pools, expected_log
 ):
     """Test that function to integrate the soil model works as expected."""
 
-    mock_integrate = mocker.patch("virtual_rainforest.models.soil.soil_model.solve_ivp")
-    mock_integrate.return_value = output
+    if mock_output:
+        mock_integrate = mocker.patch(
+            "virtual_rainforest.models.soil.soil_model.solve_ivp"
+        )
+        mock_integrate.return_value = mock_output
 
     with raises:
-        new_pools = soil_model_fixture.integrate_soil_model()
+        new_pools = soil_model_fixture.integrate()
         # Check returned pools matched (mocked) integrator output
-        assert np.allclose(new_pools["new_lmwc"], output.y[:4, -1])
-        assert np.allclose(new_pools["new_maom"], output.y[4:, -1])
+        assert np.allclose(new_pools["soil_c_pool_lmwc"], final_pools["lmwc"])
+        assert np.allclose(new_pools["soil_c_pool_maom"], final_pools["maom"])
 
     # Check that integrator is called once (and once only)
-    mock_integrate.assert_called_once()
+    if mock_output:
+        mock_integrate.assert_called_once()
 
     log_check(caplog, expected_log)
 
 
+def test_order_independance(dummy_carbon_data, soil_model_fixture):
```

Comment:
This is fine as is, but it could be part of a parametrized test, with an `order` argument that determines what order the data is in.